### PR TITLE
Ret 1452 latest

### PR DESCRIPTION
**Pull request by vinothon84**
1. Code changes for pact testing
2. build.gradle changes to pull required jars
3. Output will be created in pacts folder
4. PACT code to cover - ccd, s2s, idam & s2s api intercation
5. PACT for PDF generation

### CODE REVIEW

File: src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/CcdGetCasesByCaseIdPactTest.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+class CcdGetCasesByCaseIdPactTest extends SpringBootContractBaseTest {
+
+    private static final String TEST_CASE_ID = "1593694526480034";
+    private static final String CCD_CASE_URL = "/cases/" + TEST_CASE_ID;
+
+
+    @Pact(provider = "ccd_data_store_get_casebyid", consumer = "et_sya_api_service")
+    RequestResponsePact executeCcdGetCasesByCaseId(PactDslWithProvider builder) {
+
+        Map<String, String> responseHeaders = Map.of("Content-Type", "application/json");
```
Comment (nickwest1) [inline src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/CcdGetCasesByCaseIdPactTest.java:29]: Maybe a constant on the base class?

File: src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/StartEventForCitizenConsumerTest.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            SERVICE_AUTH_TOKEN, AUTH_TOKEN, USER_ID, "EMPLOYMENT",
+            "ET_EnglandWales", CASE_ID.toString(), UPDATE_CASE_DRAFT);
+
+        assertThat(startEventResponse.getEventId(), equalTo(UPDATE_CASE_DRAFT));
+        assertNotNull(startEventResponse.getCaseDetails());
+    }
+
+    @Override
+    @SneakyThrows
+    protected Map<String, Object> addCaseTypeJurdisticaton() {
```
Comment (nickwest1) [inline src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/StartEventForCitizenConsumerTest.java:58]: protected?

File: src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/StartForCitizenConsumerTest.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+import static io.pactfoundation.consumer.dsl.LambdaDsl.newJsonBody;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+class StartForCitizenConsumerTest extends SpringBootContractBaseTest {
+
+    @Pact(provider = "ccd_data_store_api_cases", consumer = "et_sya_api_service")
+        public RequestResponsePact startForCitizen(PactDslWithProvider builder) {
```
Comment (nickwest1) [inline src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/StartForCitizenConsumerTest.java:25]: public?

File: src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/StartForCitizenConsumerTest.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+        Map<String, String> responseHeaders = Map.of("Content-Type", "application/json");
```
Comment (nickwest1) [inline src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/StartForCitizenConsumerTest.java:27]: as constant?

File: src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/StartEventForCitizenConsumerTest.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            .append("ET_EnglandWales")
+            .append("/cases/")
+            .append(CASE_ID)
+            .append("/event-triggers/")
+            .append(UPDATE_CASE_DRAFT)
+            .append("/token")
+            .toString();
+    }
+
+    public static DslPart buildStartEventResponseWithEmptyCaseDetails(String eventId) {
```
Comment (nickwest1) [inline src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/StartEventForCitizenConsumerTest.java:80]: Duplicate maybe?

File: src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/StartForCitizenConsumerTest.java [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+            .append("EMPLOYMENT")
+            .append("/case-types/")
+            .append("ET_EnglandWales")
+            .append("/event-triggers/")
+            .append(INITIATE_CASE_DRAFT)
+            .append("/token")
+            .toString();
+    }
+
+    public static DslPart buildStartEventResponseWithEmptyCaseDetails(String eventId) {
```
Comment (nickwest1) [inline src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/StartForCitizenConsumerTest.java:78]: Duplicate maybe?
As above

File: src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/SubmitEventForCitizenConsumerTest.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+import java.util.Map;
+
+import static io.pactfoundation.consumer.dsl.LambdaDsl.newJsonBody;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+class SubmitEventForCitizenConsumerTest extends SpringBootContractBaseTest {
+
+    @Pact(provider = "ccd_data_store_api_cases", consumer = "et_sya_api_service")
+        public RequestResponsePact submitEventForCitizen(PactDslWithProvider builder) {
```
Comment (nickwest1) [inline src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/SubmitEventForCitizenConsumerTest.java:27]: public?

File: src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/SubmitEventForCitizenConsumerTest.java [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+        Map<String, String> responseHeaders = Map.of("Content-Type", "application/json");
```
Comment (nickwest1) [inline src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/SubmitEventForCitizenConsumerTest.java:29]: As constant

File: src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/SubmitEventForCitizenConsumerTest.java [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    @Test
+    @SneakyThrows
+    @PactTestFor(pactMethod = "submitEventForCitizen")
+    void verifysubmitEventForCitizen() {
+        Et1CaseData caseData = ResourceLoader.fromString(
+            "requests/caseData.json",
+            Et1CaseData.class
+        );
```
Comment (nickwest1) [inline src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/SubmitEventForCitizenConsumerTest.java:53]: Javascript pattern, Keep bracket on same line as code.
Please review all of this PR for instances of this :)

File: src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/SubmitEventForCitizenConsumerTest.java [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+        CaseDetails caseDetails = coreCaseDataApi.submitEventForCitizen(
+            SERVICE_AUTH_TOKEN, AUTH_TOKEN, USER_ID, "EMPLOYMENT",
+            "ET_EnglandWales", CASE_ID.toString(), true, caseDataContent);
+
+        assertNotNull(caseDetails);
+    }
+
+    @Override
+    @SneakyThrows
+    protected Map<String, Object> addCaseTypeJurdisticaton() {
```
Comment (nickwest1) [inline src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/SubmitEventForCitizenConsumerTest.java:70]: proitected?

File: src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/SubmitEventForCitizenConsumerTest.java [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+            .append("EMPLOYMENT")
+            .append("/case-types/")
+            .append("ET_EnglandWales")
+            .append("/cases/")
+            .append(CASE_ID)
+            .append("/events")
+            .toString();
+    }
+
+    public static DslPart buildCaseDetailsDsl(Long caseId) {
```
Comment (nickwest1) [inline src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/SubmitEventForCitizenConsumerTest.java:90]: Duplicate?
As above

File: src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/SubmitForCitizenConsumerTest.java [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+
+import java.util.Map;
+
+import static io.pactfoundation.consumer.dsl.LambdaDsl.newJsonBody;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+class SubmitForCitizenConsumerTest extends SpringBootContractBaseTest {
+
+    @Pact(provider = "ccd_data_store_api_cases", consumer = "et_sya_api_service")
+        public RequestResponsePact submitForCitizen(PactDslWithProvider builder) {
```
Comment (nickwest1) [inline src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/SubmitForCitizenConsumerTest.java:27]: public?

File: src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/SubmitForCitizenConsumerTest.java [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+        Map<String, String> responseHeaders = Map.of("Content-Type", "application/json");
```
Comment (nickwest1) [inline src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/SubmitForCitizenConsumerTest.java:28]: As above, constant

File: src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/SubmitForCitizenConsumerTest.java [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+        CaseDetails caseDetails = coreCaseDataApi.submitForCitizen(
+            SERVICE_AUTH_TOKEN, AUTH_TOKEN, USER_ID, "EMPLOYMENT",
+            "ET_EnglandWales", true, caseDataContent);
+
+        assertNotNull(caseDetails);
+    }
+
+    @Override
+    @SneakyThrows
+    protected Map<String, Object> addCaseTypeJurdisticaton() {
```
Comment (nickwest1) [inline src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/SubmitForCitizenConsumerTest.java:69]: proected?

File: src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/SubmitForCitizenConsumerTest.java [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+            .append(USER_ID)
+            .append("/jurisdictions/")
+            .append("EMPLOYMENT")
+            .append("/case-types/")
+            .append("ET_EnglandWales")
+            .append("/cases")
+            .toString();
+    }
+
+    public static DslPart buildCaseDetailsDsl(Long caseId) {
```
Comment (nickwest1) [inline src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/ccd/SubmitForCitizenConsumerTest.java:87]: Dupe?

File: src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/idam/JwksConsumerTest.java [hunk 16]
```diff
@@ -0,0 +1,79 @@
...
+            .get(mockServer.getUrl() + JWKS_AUTH_URL)
+            .then()
+            .statusCode(200)
+            .and()
+            .extract()
+            .asString();
+
+        JSONObject response = new JSONObject(responseBody);
+        Assertions.assertThat(response).isNotNull();
+        System.out.println(response);
```
Comment (nickwest1) [inline src/contractTest/java/uk/gov/hmcts/reform/et/syaapi/consumer/idam/JwksConsumerTest.java]: No System.out.println's :)

**Review by nickwest1 — CHANGES_REQUESTED** [review]
Good job in general ;)

**Review by jthmcts — APPROVED** [review]

**Pull request merged**